### PR TITLE
iPhone XR support. Better iphone X(s/+/r) support

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR fixes layout for iPhoneXR and cleans up how we deal with the non-safe areas. 
The route tree will inject a safearea above any components that aren't tagged as being underneath the notch

- [x] isIphoneX on iPhone XR
- [x] SafeAreaView in more places
- [x] remove most isIphonex
- [x] compare tabbar to master
- [x] provision
- [x] full screen attachment
- [x] profile
- [x] 5s/X/XR
- [x] storyshots once this pr is approved

### CODE REVIEW

File: shared/actions/json/config.json [hunk 1]
```diff
@@ -95,7 +95,6 @@
       "sound": "boolean",
       "writeFile": "boolean"
     },
-    "debugDump": {"items": "Array<string>"},
```
Comment (chrisnojima) [inline shared/actions/json/config.json]: this isn't used, lets just kill it

File: shared/app/global-errors/index.native.js [hunk 2]
```diff
@@ -96,36 +83,39 @@ class GlobalError extends Component<Props, State> {
...
     const {onDismiss} = this.props
     const details = this.state.cachedDetails
-    const maxHeight = GlobalError.maxHeightForSize(this.state.size)
 
     return (
-      <Box style={{...containerStyle, maxHeight}}>
-        <Box style={globalStyles.flexBoxColumn}>
-          <Box
+      <Kb.Box2 direction="vertical" style={styles.container}>
+        <Kb.NativeSafeAreaView style={{flexGrow: 0}} />
```
Comment (chrisnojima) [inline shared/app/global-errors/index.native.js]: safe area on top

File: shared/app/global-errors/index.native.js [hunk 3]
```diff
@@ -96,36 +83,39 @@ class GlobalError extends Component<Props, State> {
...
             }}
           >
-            <Text
+            <Kb.Text
               type="BodySmallSemibold"
-              style={{color: globalColors.white, flex: 1, textAlign: 'center'}}
+              style={{color: Styles.globalColors.white, flex: 1, textAlign: 'center'}}
               onClick={this._onExpandClick}
             >
+              <Kb.Icon
```
Comment (chrisnojima) [inline shared/app/global-errors/index.native.js]: i added a caret here so its more clear you can expand it

File: shared/app/global-errors/index.native.js [hunk 4]
```diff
@@ -134,73 +124,47 @@ class GlobalError extends Component<Props, State> {
...
-              style={{height: 500}}
-              windowSize={30}
-            />
-          </Box>
-        ) : (
-          <NativeScrollView>
-            <Text type="BodySmall" selectable={true} style={detailStyle}>
+          </Kb.Box>
+        </Kb.Box>
+        {this.state.size === 'Big' && (
```
Comment (chrisnojima) [inline shared/app/global-errors/index.native.js]: lets just add this if it gets big and not some static height calculation thing

File: shared/app/nav.native.js [hunk 5]
```diff
@@ -99,14 +93,14 @@ class CardStackShim extends Component<CardStackShimProps> {
...
         style={this.props.hidden ? styles.hiddenTransitioner : undefined}
-        cardStyle={{backgroundColor: globalColors.fastBlank}}
+        cardStyle={styles.card}
         transitionConfig={this._transitionConfig}
       />
     )
   }
 }
 
-const barStyle = (showStatusBarDarkContent, underStatusBar) => {
```
Comment (chrisnojima) [inline shared/app/nav.native.js]: there is no underStatusBar tag anymore

File: shared/app/nav.native.js [hunk 6]
```diff
@@ -116,41 +110,28 @@ const barStyle = (showStatusBarDarkContent, underStatusBar) => {
...
-  if (root) {
-    style = styles.sceneWrapStyleNoStatusBarPadding
-  } else {
-    style =
-      route.tags && route.tags.underStatusBar
-        ? styles.sceneWrapStyleNoStatusBarPadding
-        : styles.sceneWrapStyleWithStatusBarPadding
-  }
+  const {showStatusBarDarkContent, hideStatusBar, root, underNotch} = route.tags || {}
+  const makeSafeAreaOnTop = !root && !underNotch
```
Comment (chrisnojima) [inline shared/app/nav.native.js]: most screens get this safe area

File: shared/app/nav.native.js [hunk 7]
```diff
@@ -207,22 +188,23 @@ class MainNavStack extends Component<any, {verticalOffset: number}> {
...
-      <Box style={styles.content}>
+      <Kb.NativeView style={styles.content}>
         {stacks}
         {![chatTab].includes(props.routeSelected) ? <Offline key="offline" /> : null}
         <GlobalError key="globalError" />
-        <AnimatedTabBar show={!props.hideNav}>
-          <TabBar routeSelected={props.routeSelected} routePath={props.routePath} />
-        </AnimatedTabBar>
-      </Box>
+        <Kb.NativeSafeAreaView style={styles.tabSafeArea}>
```
Comment (chrisnojima) [inline shared/app/nav.native.js]: safe area on the bottom of the tab bar

File: shared/app/nav.native.js [hunk 8]
```diff
@@ -354,12 +336,14 @@ class Nav extends Component<Props, {keyboardShowing: boolean}> {
     const layerScreens = this.props.routeStack.filter(r => r.tags && r.tags.layerOnTop)
     const layers = layerScreens.map(r => r.leafComponent({shouldRender: true}))
 
+    // console.log('aaa', this.props, shim, layers)
+
     return (
-      <Box style={styles.shimContainer}>
+      <>
```
Comment (chrisnojima) [inline shared/app/nav.native.js]: don't actually need this extra box

File: shared/app/tab-bar/index.native.js [hunk 9]
```diff
@@ -43,18 +43,13 @@ const _selectedIconStyle = {
   color: globalColors.white,
 }
 
-const tabBarHeight = isIPhoneX ? 80 : 48
```
Comment (chrisnojima) [inline shared/app/tab-bar/index.native.js]: 🔥

File: shared/app/tab-bar/index.stories.js [hunk 10]
```diff
@@ -36,7 +36,6 @@ const containerStyle = platformStyles({
     ...globalStyles.flexBoxRow,
   },
   isMobile: {
-    marginTop: 40, // Avoid the notch on iPhoneX
```
Comment (chrisnojima) [inline shared/app/tab-bar/index.stories.js]: 🔥

File: shared/chat/conversation/attachment-fullscreen/index.native.js [hunk 11]
```diff
@@ -55,32 +45,32 @@ class AutoMaxSizeImage extends Component<any, {width: number, height: number, lo
...
-          ...globalStyles.fillAbsolute,
-          backgroundColor: globalColors.black,
-          paddingTop: isIPhoneX ? globalMargins.medium : undefined,
+          ...Styles.globalStyles.flexBoxColumn,
+          ...Styles.globalStyles.fillAbsolute,
+          backgroundColor: Styles.globalColors.black,
         }}
       >
-        <Text
+        <Kb.NativeSafeAreaView style={{backgroundColor: Styles.globalColors.black, flexGrow: 0}} />
```
Comment (chrisnojima) [inline shared/chat/conversation/attachment-fullscreen/index.native.js]: under notch black area

File: shared/chat/conversation/attachment-fullscreen/index.native.js [hunk 12]
```diff
@@ -105,18 +95,19 @@ class _Fullscreen extends React.Component<Props & OverlayParentProps, {loaded: b
           position="bottom left"
           visible={this.props.showingMenu}
         />
-      </Box>
+        <Kb.NativeSafeAreaView style={{backgroundColor: Styles.globalColors.black, flexGrow: 0}} />
```
Comment (chrisnojima) [inline shared/chat/conversation/attachment-fullscreen/index.native.js]: bottom black area

File: shared/chat/conversation/attachment-get-titles/index.js [hunk 13]
```diff
@@ -194,4 +180,4 @@ const stylesInputMobile = {
   paddingRight: 20,
 }
 
-export default GetTitles
+export default Kb.HeaderOrPopup(GetTitles)
```
Comment (chrisnojima) [inline shared/chat/conversation/attachment-get-titles/index.js]: on master looks pretty crappy, its a popup so it looks weirdly squished, instead just make this a regular route

File: shared/constants/platform.native.js [hunk 14]
```diff
@@ -30,7 +30,7 @@ export const isAndroidNewerThanM = isAndroid && parseInt(mobileOsVersion) > 22
 export const isAndroidNewerThanN = isAndroid && parseInt(mobileOsVersion, 10) >= 26
 
 export const isIPhoneX =
-  Platform.OS === 'ios' && !Platform.isPad && !Platform.isTVOS && Dimensions.get('window').height === 812
+  Platform.OS === 'ios' && !Platform.isPad && !Platform.isTVOS && Dimensions.get('window').height >= 812
```
Comment (chrisnojima) [inline shared/constants/platform.native.js]: support iphone xr

File: shared/chat/routes.js [hunk 15]
```diff
@@ -69,19 +69,19 @@ const chatChildren = {
   },
   attachmentFullscreen: {
     component: AttachmentFullscreen,
-    tags: makeLeafTags(isMobile ? {hideStatusBar: true, fullscreen: true} : {layerOnTop: true}),
+    tags: makeLeafTags(
+      isMobile ? {hideStatusBar: true, fullscreen: true, underNotch: true} : {layerOnTop: true}
+    ),
     children: key => makeRouteDefNode(chatChildren[key]),
   },
   attachmentVideoFullscreen: {
```
Comment (buoyad) [inline shared/chat/routes.js]: Is there any way to reach this on mobile?

File: shared/chat/conversation/attachment-fullscreen/index.native.js [hunk 16]
```diff
@@ -55,32 +45,31 @@ class AutoMaxSizeImage extends Component<any, {width: number, height: number, lo
...
 }
 
-class _Fullscreen extends React.Component<Props & OverlayParentProps, {loaded: boolean}> {
+class _Fullscreen extends React.Component<Props & Kb.OverlayParentProps, {loaded: boolean}> {
   state = {loaded: false}
   _setLoaded = () => this.setState({loaded: true})
   render() {
     return (
-      <Box
+      <Kb.NativeSafeAreaView
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/chat/conversation/attachment-fullscreen/index.native.js]: Whoops nevermind, I see there is one

File: shared/chat/routes.js (hunk 15 above)
Comment (chrisnojima) [reply shared/chat/routes.js]: discuss w/ @mmaxim and we think not. i'll try and remove it

Comment (mmaxim) [reply shared/chat/routes.js]: Before the inline player was a thing, on Android it would show this, but I think that is gone now.

**Comment by chrisnojima** [comment]
- [ ] anroid

**Comment by chrisnojima** [comment]
@buoyad this handles android now. The android impl of SafeAreaView is effectively a blank <View>

File: shared/chat/routes.js (hunk 15 above)
Comment (chrisnojima) [reply shared/chat/routes.js]: i removed it

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/app/nav.native.js [hunk 17]
```diff
@@ -116,41 +110,32 @@ const barStyle = (showStatusBarDarkContent, underStatusBar) => {
   if (showStatusBarDarkContent) {
```
Comment (buoyad) [inline shared/app/nav.native.js]: Maybe we should change the semantics of this to `showStatusBarLightContent`. there's no way to set the content to light on iOS. we'd want to do that on the attachment fullscreen page.

**Pull request merged**